client/v2/client.go

```diff
@@ -413,16 +414,45 @@ func (uc *udpclient) Write(bp BatchPoints) error {
 	for _, p := range bp.Points() {
 		pointstring := p.pt.RoundedString(d) + "\n"
 
-		// Write and reset the buffer if we reach the max size
 		if b.Len()+len(pointstring) >= uc.payloadSize {
 			if _, err := uc.conn.Write(b.Bytes()); err != nil {
 				return err
 			}
 			b.Reset()
 		}
 
-		if _, err := b.WriteString(pointstring); err != nil {
-			return err
+		if b.Len()+len(pointstring) <= uc.payloadSize {
+			if _, err := b.WriteString(pointstring); err != nil {
+				return err
+			}
+		} else {
+			if p.Time().IsZero() {
+				if len(pointstring) > MaxPointSize {
+					return fmt.Errorf("point size is greater than %dB", MaxPointSize)
```

Comment:
Also, this should probably be package-level variable: `var ErrLargePoint = errors.New("point exceeds allowed size")`. This also raises the question about what to do when a single point in a batch is too large... If this point is the last one, then the earlier points will have been sent. Perhaps the point should be dropped, and an error stored indicating some points were dropped, and try to send the rest?

@jwilder, what do you think?